web/src/main/java/com/futurewei/alcor/web/entity/node/NodeInfo.java

```diff
@@ -57,27 +57,33 @@
     @JsonProperty("group_topic")
     private String groupTopic;
 
+    @JsonProperty("ncm_uri")
+    private String ncm_uri;
+
+    @JsonProperty("ncm_id")
+    private String ncm_id;
+
     public NodeInfo() {
 
     }
 
     public NodeInfo(NodeInfo nodeInfo) {
-        this(nodeInfo.id, nodeInfo.name, nodeInfo.localIp, nodeInfo.macAddress, nodeInfo.veth, nodeInfo.gRPCServerPort, nodeInfo.hostDvrMac, nodeInfo.unicastTopic, nodeInfo.multicastTopic, nodeInfo.groupTopic);
+        this(nodeInfo.id, nodeInfo.name, nodeInfo.localIp, nodeInfo.macAddress, nodeInfo.veth, nodeInfo.gRPCServerPort, nodeInfo.hostDvrMac, nodeInfo.unicastTopic, nodeInfo.multicastTopic, nodeInfo.groupTopic, nodeInfo.ncm_uri, nodeInfo.ncm_id);
     }
 
-    public NodeInfo(String id, String name, String localIp, String macAddress, String veth, int gRPCServerPort, String unicastTopic, String multicastTopic, String groupTopic) {
-        this(id, name, localIp, macAddress, unicastTopic, multicastTopic, groupTopic);
+    public NodeInfo(String id, String name, String localIp, String macAddress, String veth, int gRPCServerPort, String unicastTopic, String multicastTopic, String groupTopic, String ncm_uri, String ncm_id) {
+        this(id, name, localIp, macAddress, unicastTopic, multicastTopic, groupTopic, ncm_uri, ncm_id);
         this.veth = veth;
         this.gRPCServerPort = gRPCServerPort;
     }
 
-    public NodeInfo(String nodeId, String nodeName, String ipAddress, String macAddress, int gRPCServerPort, String unicastTopic, String multicastTopic, String groupTopic) {
-        this(nodeId, nodeName, ipAddress, macAddress, unicastTopic, multicastTopic, groupTopic);
+    public NodeInfo(String nodeId, String nodeName, String ipAddress, String macAddress, int gRPCServerPort, String unicastTopic, String multicastTopic, String groupTopic, String ncm_uri, String ncm_id) {
+        this(nodeId, nodeName, ipAddress, macAddress, unicastTopic, multicastTopic, groupTopic, ncm_uri, ncm_id);
```

Comment:
@pkommoju I think you can create a new overloaded constructor NodeInfo(....) to include all parameters and two new parameters you added. Don't edit any existing constructors. Inside the constructor, you can use this() parent constructor to initialize the matching parameters and use another two statements to assign values for two new parameters.

Reply:
+1

Reply:
James,
I am looking into this.

Reply:
After going through all calls to NodeInfo constructor, I realized that most variations of constructors are inside NodeInfo itself, which will be taken care by your suggestion, in addition to changing this constructor, which is sort of common constructor to set ncm_uri and ncm_id to null.

>>> public NodeInfo(String nodeId, String nodeName, String ipAddress, String macAddress, String unicastTopic, String multicastTopic, String groupTopic)

But then there are these other cases from outside NodeInfo:
NodeFileLoader, NodeControllerTest, MockRestClientAndRepository all use

NodeInfo(strId, strName, strIp, strMac, strVeth, nServerPort, strUnicastTopic, strMulticastTopic, strGroupTopic);

They need to pass ncm_uri, ncm_id. So I had to add another override for this constructor.

I have a question about style/convention here. Is it preferred to set them to "" or null?

I need to refresh my branch and build and test.

Reply:
You just need to create a new constructor like:
```
public NodeInfo(String nodeId, String nodeName, String ipAddress, String macAddress, int gRPCServerPort, String unicastTopic, String multicastTopic, String groupTopic, String ncm_uri, String ncm_id) {
        this(nodeId, nodeName, ipAddress, macAddress, unicastTopic, multicastTopic, groupTopic);
        this.ncm_uri = ncm_uri;
        this.ncm_id = ncm_id;
}
```
You don't need to modify other constructors, I think.

Reply:
If the basecase like constructor is not modifed, what will the values of ncm_uri and ncm_id? Won't they be garbage?

Reply:
if there is no initialization for a data member in the constructor, that data member will be null. You can use set method to set value for data members too.

Reply:
Well, then, C++ is still lingering in my head. I will get rid of the other constructor changes.